tasks/setup_plugins.yml

```diff
@@ -21,6 +21,8 @@
     update: True
   tags: [ 'role::checkmk_agent:plugins:get' ]
   delegate_to: '{{ checkmk_agent__git_dest_host }}'
+  become: '{{ False if (checkmk_agent__git_dest_host in ["localhost"]) else True }}'
+  become_user: '{{ lookup("env","USER") if (checkmk_agent__git_dest_host in ["localhost"]) else ansible_become_user|d("root") }}'
```

Comment:
Why you explicitly define `become_user`? When `become: True` (default, inherited by the playbook), everything works as before. When `become: False`, `become_user` is ignored.